services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/tenant/AutowiredTenantInformationService.java

```diff
@@ -53,6 +55,12 @@
                 });
     }
 
+    @Override
+    public Future<TenantResult<TenantObject>> getTenant(final String tenantId, final Span span) {
+        return service.get(tenantId, span)
+                .map(tenantJson -> TenantResult.from(HttpURLConnection.HTTP_OK, tenantJson.getPayload().mapTo(TenantObject.class)));
```

Comment:
this will only work if the result has status 200 and thus the payload contains JSON .... see implementation of *tenantExists* ...

Reply:
True. 

FMPOV a failed Future would make more sense (in contrast to the implementation of _tenantExists_). Otherwise every caller of the _getTenant_ method would need to start with an _if_ examining the result.

In general I find this pattern (returning a result which is possibly an error) cumbersome - isn't that what the `onFailure` methods in a Future are meant for?

Reply:
yes, and we all agree that we need to change the service interfaces accordingly. But we will definitely not have some methods in some interfaces follow one approach while others follow another one. At least we want to be consistent ...

Reply:
Fixed (in a consistent way ;-) )